src/application/application.cpp

```diff
@@ -226,12 +226,23 @@ void Application::Predict() {
                       config_.precise_float_parser);
     TextReader<int> result_reader(config_.output_result.c_str(), false);
     result_reader.ReadAllLines();
-    std::vector<std::vector<int>> pred_leaf(result_reader.Lines().size());
+
+    auto nrow = result_reader.Lines().size();
+    auto ncol = 0;
+    if (nrow > 0) {
+      ncol = result_reader.Lines()[0].size();
+    }
+    std::vector<int> pred_leaf;
+    pred_leaf.reserve(nrow * ncol);
+
     #pragma omp parallel for num_threads(OMP_NUM_THREADS()) schedule(static)
-    for (int i = 0; i < static_cast<int>(result_reader.Lines().size()); ++i) {
-      pred_leaf[i] = Common::StringToArray<int>(result_reader.Lines()[i], '\t');
+    for (size_t irow = 0; irow < nrow; ++irow) {
+      auto line_vec = Common::StringToArray<int>(result_reader.Lines()[irow], '\t');
+      for (int i_row_item = 0; i_row_item < ncol; ++i_row_item) {
+	pred_leaf[i_row_item] = line_vec[i_row_item];
```

Comment:
Doesn't this need to be

```suggestion
	pred_leaf[irow * ncol + i_row_item] = line_vec[i_row_item];
```